drivers/docker/driver.go

```diff
@@ -489,24 +501,27 @@ CREATE:
 // createContainerClient is the subset of Docker Client methods used by the
 // createContainer method to ease testing subtle error conditions.
 type createContainerClient interface {
-	CreateContainer(docker.CreateContainerOptions) (*docker.Container, error)
-	InspectContainer(id string) (*docker.Container, error)
-	ListContainers(docker.ListContainersOptions) ([]docker.APIContainers, error)
-	RemoveContainer(opts docker.RemoveContainerOptions) error
+	ContainerCreate(context.Context, *containerapi.Config, *containerapi.HostConfig, *networkapi.NetworkingConfig, *ocispec.Platform, string) (containerapi.CreateResponse, error)
+	ContainerInspect(context.Context, string) (types.ContainerJSON, error)
+	ContainerList(context.Context, containerapi.ListOptions) ([]types.Container, error)
+	ContainerRemove(context.Context, string, containerapi.RemoveOptions) error
 }
 
 // createContainer creates the container given the passed configuration. It
 // attempts to handle any transient Docker errors.
-func (d *Driver) createContainer(client createContainerClient, config docker.CreateContainerOptions,
-	image string) (*docker.Container, error) {
+func (d *Driver) createContainer(client createContainerClient, config createContainerOptions, image string) (*types.ContainerJSON, error) {
 	// Create a container
 	var attempted uint64
 	var backoff time.Duration
 
 CREATE:
-	container, createErr := client.CreateContainer(config)
+	_, createErr := client.ContainerCreate(d.ctx, config.Config, config.Host, config.Networking, nil, config.Name)
```

Comment:
Looks like the missing arg here is [`ocispec.Platform`](https://pkg.go.dev/github.com/opencontainers/image-spec/specs-go/v1#Platform). Do we know if that's at all useful for Docker?

Reply:
We don't use this parameter at the moment in Nomad, but it could be useful in the future. It allows specifying platform of created containers.